real_preprocessing/src/opt_visualization.cpp

```diff
@@ -0,0 +1,221 @@
+}
+
+geometry_msgs::Pose getCampose(int filenum)
+{
+	YAML::Node YAML_handle = YAML::LoadFile(getDetfilePath(filenum));
+
+	geometry_msgs::Pose cam_pose;
+	cam_pose.position.x = YAML_handle["world_T_camera"]["translation"][0].as<double>();
+	cam_pose.position.y = YAML_handle["world_T_camera"]["translation"][1].as<double>();
+	cam_pose.position.z = YAML_handle["world_T_camera"]["translation"][2].as<double>();
+
+	double rod_vec[3] = {YAML_handle["world_T_camera"]["rotation"][0].as<double>(),
+							YAML_handle["world_T_camera"]["rotation"][1].as<double>(),
+							YAML_handle["world_T_camera"]["rotation"][2].as<double>()};
+
+	Eigen::Quaterniond quat = rodrigToQuat(rod_vec);
+	cam_pose.orientation.x = quat.x();
+	cam_pose.orientation.y = quat.y();
+	cam_pose.orientation.z = quat.z();
+	cam_pose.orientation.w = quat.w();
+
+	return cam_pose;
+}
+
+//populates camera marker with world_T_cam from given file number
+void camUpdate(int filenum, visualization_msgs::Marker& cam_marker, geometry_msgs::Pose& cam_pose)
+{
+	cam_pose = getCampose(filenum);
+
+	cam_marker.header.frame_id = "world"; 
+	cam_marker.header.stamp = ros::Time();
+	cam_marker.ns = "camera";
+	cam_marker.id = -1;
+	cam_marker.type = visualization_msgs::Marker::CUBE;
+	cam_marker.action = visualization_msgs::Marker::ADD;
+	cam_marker.pose = cam_pose;
+	cam_marker.scale.x = 0.08;
+	cam_marker.scale.y = 0.08;
+	cam_marker.scale.z = 0.04;
+	cam_marker.color.a = 1.0; 
+	cam_marker.color.r = 1.00;
+	cam_marker.color.g = 0.41;
+	cam_marker.color.b = 0.71;
+	cam_marker.lifetime = ros::Duration();
+}
+
+int main(int argc, char **argv)
+{
+	ros::init(argc, argv, "opt_visualization");
+	ros::NodeHandle nh;
+	ros::Publisher tag_pub = nh.advertise<visualization_msgs::MarkerArray>("tag_markers", 1000);
+	ros::Publisher tag_ps_pub = nh.advertise<geometry_msgs::PoseArray>("world_T_tags_ps", 1000);
+	ros::Publisher cam_pub = nh.advertise<visualization_msgs::Marker>("cam_vis_marker", 1000);
+	ros::Publisher cam_ps_pub = nh.advertise<geometry_msgs::PoseArray>("world_T_cam_ps", 1000);
+	ros::Publisher cam_traj_pub = nh.advertise<visualization_msgs::Marker>("cam_traj", 1000);
+	ros::Rate loop_rate(5);
+	
+	int filenum(0);
+	visualization_msgs::MarkerArray tag_array;
+	visualization_msgs::Marker cam_marker;
+	visualization_msgs::Marker traj_marker;
+	geometry_msgs::PoseArray tagpose_array;
+	geometry_msgs::PoseArray campose_array;
+
+	tagpose_array.header.frame_id = "world";
+	campose_array.header.frame_id = "world";
+
+	// Publish the world_T_tags from the targets.yaml
+	YAML::Node YAML_handle= YAML::LoadFile(getTargetpath());
+	for(int tag_index = 0; tag_index != YAML_handle["targets"].size(); tag_index++) 
+	{
+		geometry_msgs::Pose tag_pose;
+		visualization_msgs::Marker tag_marker;
+		loadTagmarker(tag_index, tag_marker, tag_pose);
+
+		tag_array.markers.push_back(tag_marker);
+		tagpose_array.poses.push_back(tag_pose);
+	}
+
+	// Continuously publish the world_T_cam for each detection file
```

Comment:
Continuously publish the world_T_cam for each detection file and all world_T_tags.